checks/src/main/kotlin/com/jzbrooks/assertk/lint/checks/KotlinAssertionDetector.kt

```diff
@@ -0,0 +1,86 @@
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.UElement
+import java.util.EnumSet
+
+class KotlinAssertionDetector :
+    Detector(),
+    Detector.UastScanner {
+    override fun getApplicableUastTypes(): List<Class<out UElement>> =
+        listOf(
+            UCallExpression::class.java,
+        )
+
+    override fun createUastHandler(context: JavaContext) =
+        object : UElementHandler() {
+            override fun visitCallExpression(node: UCallExpression) {
+                if (!node.isKotlin) return
+
+                val method = node.resolve() ?: return
+                val containingClass = method.containingClass?.qualifiedName ?: return
+                if (method.name == "assert" &&
+                    (
+                        containingClass == "kotlin.PreconditionsKt" ||
+                            containingClass == "kotlin.PreconditionsKt__AssertionsJVMKt"
+                    )
+                ) {
+                    context.report(
+                        ISSUE,
+                        node,
+                        context.getCallLocation(
+                            node,
+                            includeReceiver = false,
+                            includeArguments = true,
+                        ),
+                        ISSUE.getBriefDescription(TextFormat.TEXT),
+                        fix()
+                            .replace()
+                            .reformat(true)
+                            .shortenNames()
+                            .range(
+                                context.getCallLocation(
+                                    node,
+                                    includeReceiver = false,
+                                    includeArguments = true,
+                                ),
+                            ).imports("assertk.assertThat", "assertk.assertions.isTrue")
+                            .with(
+                                "assertThat(${node.getArgumentForParameter(
+                                    0,
+                                )!!.sourcePsi!!.text}).isTrue()",
+                            ).build(),
+                    )
+                }
+            }
+        }
+
+    companion object {
+        @JvmField
+        val ISSUE: Issue =
+            Issue.create(
+                id = "KotlinAssertionUse",
+                briefDescription = "Kotlin assertion is used",
+                explanation = """
+                    assertk assertions are preferred over kotlin assertions for more descriptive and consistent error messages.
+                """,
+                category = Category.CORRECTNESS,
+                priority = 6,
+                severity = Severity.WARNING,
+                implementation =
+                    Implementation(
+                        KotlinAssertionDetector::class.java,
+                        EnumSet.of(Scope.JAVA_FILE, Scope.TEST_SOURCES),
```

Comment:
This scope config shouldn't bother kotlin.assert calls in non test code. Every scope in this set has to apply for the detector to run.

https://googlesamples.github.io/android-custom-lint-rules/api-guide.html#toc2.7